packages/eslint-plugin/rules/__tests__/a11y-anchor-has-content.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * External dependencies
+ */
+import { RuleTester } from 'eslint';
+
+/**
+ * Internal dependencies
+ */
+import rule from '../a11y-anchor-has-content';
+
+const ruleTester = new RuleTester( {
+	parserOptions: {
+		sourceType: 'module',
+		ecmaVersion: 6,
+		ecmaFeatures: {
+			jsx: true,
+		},
+	},
+} );
+
+ruleTester.run( 'a11y-anchor-has-content', rule, {
+	valid: [
+		{
+			code: `
+			( ) => {
+				( <div>
+					   { createInterpolateElement( __( 'This is a <a> link </a>.' ), {
+							   a: <a href="https://wordpress.org" />,
+					   } ) }
+				   </div>
+			   )
+		   };`,
+		},
+		{
+			code: `
+			( ) => {
+				( <div>
+					   { createInterpolateElement( __( 'help me <a> test with multiple words </a>.' ), {
+							   a: <a href="https://example.com" target="_blank" />,
+					   } ) }
+				   </div>
+			   )
+		   };`,
+		},
+		// Anchor with a custom component as content. It may or may not be valid. No errors are triggered.
+		{
+			code: `
+			( ) => {
+				( <div>
+					   { createInterpolateElement( __( 'help me <a> <TextWrapper /> </a>.' ), {
+							   a: <a href="https://example.com" target="_blank" />,
+					   } ) }
+				   </div>
+			   )
+		   };`,
+		},
+		{
+			// Anchor without createInterpolateElement
+			code: `
+			( ) => {
+				( <div>
+					   <a href="https://example.com"> test with multiple words </a>.
+				   </div>
+			   )
+		   };`,
+		},
+		{
+			// ⚠️ This is invalid code that would get flagged by jsx-a11y/anchor-is-valid, but NOT by eslint-plugin-jsx-a11y/anchor-has-content or @wordpress/a11-anchor-has-content.
```

Comment:
It’s also possible to add names for tests according to https://eslint.org/docs/latest/integrate/nodejs-api#ruletester. Maybe that would be a way to provide additional context instead of using inline comments.

Reply:
Thank you so much, completely missed the `name` field. I've switched to using name in most cases.